storage/remote/queue_manager.go

```diff
@@ -541,7 +541,7 @@ func (t *QueueManager) StoreSeries(series []record.RefSeries, index int) {
 		ls := processExternalLabels(s.Labels, t.externalLabels)
 		lbls := relabel.Process(ls, t.relabelConfigs...)
 		if len(lbls) == 0 {
-			t.droppedSeries[s.Ref] = struct{}{}
+			t.droppedSeries[s.Ref] = index
 			continue
 		}
 		t.seriesSegmentIndexes[s.Ref] = index
```

Comment:
The idea of `seriesSegmentIndexes` is that it keeps track of all refs -> index mappings. I think all that needs to be done in this PR is move 
```
t.seriesSegmentIndexes[s.Ref] = index
```
to the beginning of the `for` loop. Then the loop in `SeriesReset` will also properly work without needing to change `droppedSeries` to an int from a struct. It is probably worth adding a comment as well.

Reply:
Sure, I will make the other PL to fix this one. Thanks for your advice.